packages/core/mesh/messaging/src/signal-manager/websocket-signal-manager.ts

```diff
@@ -142,26 +145,35 @@ export class WebsocketSignalManager implements SignalManager {
     log('signal', { recipient });
     invariant(this._opened, 'Closed');
 
-    void this._forEachServer(async (server, serverName) => {
-      void server.sendMessage({ author, recipient, payload }).catch((err) => {
-        if (err instanceof RateLimitExceededError) {
-          log.info('WSS rate limit exceeded', { err });
-        } else if (err instanceof TimeoutError || err.constructor.name === 'TimeoutError') {
-          log.info('WSS sendMessage timeout', { err });
-          void this.checkServerFailure(serverName);
-        } else {
-          log.info(`error sending to ${serverName}`, { err });
-          void this.checkServerFailure(serverName);
-        }
-      });
+    void this._forEachServer(async (server, serverName, index) => {
+      void server
+        .sendMessage({ author, recipient, payload })
+        .then(() => this._clearServerFailedFlag(serverName, index))
+        .catch((err) => {
+          if (err instanceof RateLimitExceededError) {
+            log.info('WSS rate limit exceeded', { err });
+            this._monitor.recordRateLimitExceeded();
+          } else if (err instanceof TimeoutError || err.constructor.name === 'TimeoutError') {
+            log.info('WSS sendMessage timeout', { err });
+            void this.checkServerFailure(serverName, index);
```

Comment:
Should this try and distinguish application or RPC-level failures (e.g. software failure on the server side) with network-level errors (e.g. loss of Internet connectivity)?